src/annotator/integrations/test/vitalsource-test.js

```diff
@@ -0,0 +1,215 @@
+        'content'
+      );
+    });
+
+    it('returns `null` if the book container element is not found', () => {
+      fakeViewer.destroy();
+      assert.isNull(vitalSourceFrameRole());
+    });
+  });
+
+  describe('VitalSourceContainerIntegration', () => {
+    let fakeGuest;
+    let integration;
+
+    beforeEach(() => {
+      fakeGuest = {
+        injectClient: sinon.stub(),
+      };
+      integration = new VitalSourceContainerIntegration(fakeGuest);
+    });
+
+    afterEach(() => {
+      integration.destroy();
+    });
+
+    it('throws if constructed outside the VitalSource book reader', () => {
+      fakeViewer.destroy();
+      assert.throws(() => {
+        new VitalSourceContainerIntegration(fakeGuest);
+      }, 'Book container element not found');
+    });
+
+    it('injects client into content frame', () => {
+      assert.calledWith(fakeGuest.injectClient, fakeViewer.contentFrame);
+    });
+
+    it('re-injects client when content frame is changed', async () => {
+      fakeGuest.injectClient.resetHistory();
+
+      fakeViewer.loadNextChapter();
+      await delay(0);
+
+      assert.calledWith(fakeGuest.injectClient, fakeViewer.contentFrame);
+    });
+
+    it('does not allow annotation in the container frame', async () => {
+      assert.equal(integration.canAnnotate(), false);
+
+      // Briefly check the results of the stub methods.
+      assert.instanceOf(await integration.anchor(), Range);
+      assert.throws(() => integration.describe());
+      assert.equal(integration.fitSideBySide(), false);
+      assert.deepEqual(await integration.getMetadata(), {
+        title: '',
+        link: [],
+      });
+      assert.equal(await integration.uri(), document.location.href);
+      assert.equal(integration.contentContainer(), document.body);
+      await integration.scrollToAnchor({}); // nb. No assert, this does nothing.
+    });
+  });
+
+  describe('VitalSourceContentIntegration', () => {
+    let integration;
+
+    beforeEach(() => {
+      integration = new VitalSourceContentIntegration();
+    });
+
+    afterEach(() => {
+      integration.destroy();
+    });
+
+    it('allows annotation', () => {
+      assert.equal(integration.canAnnotate(), true);
+    });
+
+    it('does not support side-by-side mode', () => {
+      assert.equal(integration.fitSideBySide(), false);
+    });
+
+    it('stops mouse events from propagating to parent frame', () => {
+      const events = ['mousedown', 'mouseup', 'mouseout'];
+
+      for (let eventName of events) {
+        const listener = sinon.stub();
+        document.addEventListener(eventName, listener);
+
+        const event = new Event(eventName, { bubbles: true });
+        document.body.dispatchEvent(event);
+        assert.notCalled(listener);
+
+        document.removeEventListener(eventName, listener);
+      }
+    });
+
+    it('delegates to HTML integration for anchoring', async () => {
+      await integration.contentContainer();
```

Comment:
It is likely I'm overlooking something, but neither integration (`VitalSource` nor `HTML`) implements `contentContainer` as `async`. The `Implementation` interface doesn't seem to suggest it, either.

Reply:
Oh, indeed. You're right.